modules/miner/mine.go

```diff
@@ -45,6 +41,64 @@ func (m *Miner) blockForWork() (b types.Block) {
 	return
 }
 
+// solveBlock takes a block, target, and number of iterations as input and
+// tries to find a block that meets the target. This function can take a long
+// time to complete, and should not be called with a lock.
+func (m *Miner) solveBlock(blockForWork types.Block, target types.Target, iterations uint64) (b types.Block, solved bool, err error) {
+	b = blockForWork
+	bRoot := b.MerkleRoot()
+	hashbytes := make([]byte, 72)
+	copy(hashbytes, b.ParentID[:])
+	copy(hashbytes[40:], bRoot[:])
+
+	// Iterate through a bunch of nonces (from a random starting point) and try
+	// to find a winnning solution.
+	nonce := (*uint64)(unsafe.Pointer(&hashbytes[32]))
+	*nonce = b.Nonce
+	for i := 0; i < int(iterations); i++ {
+		*nonce++
+		id := crypto.HashBytes(hashbytes)
+		if bytes.Compare(target[:], id[:]) >= 0 {
+			b.Nonce = binary.LittleEndian.Uint64(hashbytes[32:])
```

Comment:
b.Nonce is still a uint64 eh?

Should probably specify `hashbytes[32:40]` just for clarity's sake.